trace/redis/redis.go

```diff
@@ -0,0 +1,79 @@
+package redis
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/beatlabs/patron/trace"
+	"github.com/go-redis/redis/v7"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+const (
+	// RedisComponent definition
+	RedisComponent = "redis"
+	// RedisDBType description
+	RedisDBType = "In-memory"
+)
+
+// Span starts a new Redis child span with specified tags.
+func Span(ctx context.Context, opName, cmp, dbType, instance, stmt string,
+	tags ...opentracing.Tag) (opentracing.Span, context.Context) {
+
+	sp, ctx := opentracing.StartSpanFromContext(ctx, opName)
+	ext.Component.Set(sp, cmp)
+	ext.DBType.Set(sp, dbType)
+	ext.DBInstance.Set(sp, instance)
+	ext.DBStatement.Set(sp, stmt)
+	for _, t := range tags {
+		sp.SetTag(t.Key, t.Value)
+	}
+	return sp, ctx
+}
+
+// Options wraps redis.Options for easier usage.
+type Options redis.Options
+
+// Nil represents the error which is returned in case a key is not found.
+const Nil = redis.Nil
+
+// Client represents a connection with a Redis client.
+type Client struct {
+	*redis.Client
+}
+
+func (c *Client) startSpan(ctx context.Context, opName, stmt string) (opentracing.Span, context.Context) {
```

Comment:
we have two methods here Span and startSpan... do we need both? Can we have only the startSpan one?

Reply:
Yes, you're right. Since the span function now lives here and not in `trace.go`, they're not both required. I have kept only startSpan, but this cleanup will happen in other packages as well as part of #145.